flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-iceberg/src/main/java/org/apache/flink/cdc/connectors/iceberg/sink/IcebergMetadataApplier.java

```diff
@@ -0,0 +1,308 @@
+                    return null;
+                },
+                truncateTableEvent -> {
+                    throw new UnsupportedSchemaChangeEventException(truncateTableEvent);
+                });
+    }
+
+    private void applyCreateTable(CreateTableEvent event) {
+        try {
+            TableIdentifier tableIdentifier = TableIdentifier.parse(event.tableId().identifier());
+
+            // Step1: Build Schema.
+            org.apache.flink.cdc.common.schema.Schema cdcSchema = event.getSchema();
+            List<Types.NestedField> columns = new ArrayList<>();
+            Set<Integer> identifierFieldIds = new HashSet<>();
+            for (int index = 0; index < event.getSchema().getColumnCount(); index++) {
+                columns.add(
+                        IcebergTypeUtils.convertCDCColumnToIcebergField(
+                                index, (PhysicalColumn) cdcSchema.getColumns().get(index)));
+                if (cdcSchema.primaryKeys().contains(cdcSchema.getColumns().get(index).getName())) {
+                    identifierFieldIds.add(index);
+                }
+            }
+
+            // Step2: Build partition spec.
+            Schema icebergSchema = new Schema(columns, identifierFieldIds);
+            List<String> partitionColumns = cdcSchema.partitionKeys();
+            if (partitionMaps.containsKey(event.tableId())) {
+                partitionColumns = partitionMaps.get(event.tableId());
+            }
+            PartitionSpec.Builder builder = PartitionSpec.builderFor(icebergSchema);
+            for (String name : partitionColumns) {
+                // TODO Add more partition transforms, see
+                // https://iceberg.apache.org/spec/#partition-transforms.
+                builder.identity(name);
+            }
+            PartitionSpec partitionSpec = builder.build();
+            if (!catalog.tableExists(tableIdentifier)) {
+                catalog.createTable(tableIdentifier, icebergSchema, partitionSpec, tableOptions);
+            }
+        } catch (Exception e) {
+            throw new SchemaEvolveException(event, e.getMessage(), e);
+        }
+    }
+
+    private void applyAddColumn(AddColumnEvent event) {
+        TableIdentifier tableIdentifier = TableIdentifier.parse(event.tableId().identifier());
+        try {
+            Table table = catalog.loadTable(tableIdentifier);
+            applyAddColumnEventWithPosition(table, event);
+        } catch (Exception e) {
+            throw new SchemaEvolveException(event, e.getMessage(), e);
+        }
+    }
+
+    private void applyAddColumnEventWithPosition(Table table, AddColumnEvent event)
+            throws SchemaEvolveException {
+
+        try {
+            UpdateSchema updateSchema = table.updateSchema();
+            for (AddColumnEvent.ColumnWithPosition columnWithPosition : event.getAddedColumns()) {
+                Column addColumn = columnWithPosition.getAddColumn();
```

Comment:
```suggestion
                Column addColumn = columnWithPosition.getAddColumn();
               String columnName = addColumn.getName();
               LogicalType logicalType = 
                                FlinkSchemaUtil.convert(
                                        DataTypeUtils.toFlinkDataType(addColumn.getType())
                                                .getLogicalType());
```
Use `columnName` and `logicalType` in following code.